examples/csv-report/src/main/java/org/gauge/CsvReporter.java

```diff
@@ -1,94 +1,24 @@
+import java.util.concurrent.Executors;
 
 public class CsvReporter {
-    private static final String LOCALHOST = "127.0.0.1";
-    private static final String ReportDir = "csv-report-example";
-    private static final String ResultFileName = "result.csv";
-
-    public static void main(String[] args){
-        String portEnv = System.getenv("plugin_connection_port");
-        int port = Integer.parseInt(portEnv);
-        Socket socket;
-        while (true) {
-            try {
-                socket = new Socket(LOCALHOST, port);
-                break;
-            } catch (Exception ignored) {
-            }
-        }
-        
-        try {
-            while (!socket.isClosed() && socket.isConnected()) {
-                Message message = Message.parseDelimitedFrom(socket.getInputStream());
-                if(message.getMessageType() == Message.MessageType.SuiteExecutionResult) {
-                    SuiteExecutionResult result = message.getSuiteExecutionResult();
-                    FileWriter writer = new FileWriter(getTargetFile());
-                    writer.write("Spec,Scenario,Result,Time Taken\n");
-                    for (ProtoSpecResult spec : result.getSuiteResult().getSpecResultsList()) {
-                        for (ProtoItem item : spec.getProtoSpec().getItemsList()) {
-                            if(item.getItemType() == ItemType.Scenario){
-                                ProtoScenario scenario = item.getScenario();
-                                String s = spec.getProtoSpec().getSpecHeading() + "," +
-                                    scenario.getScenarioHeading() + "," +
-                                    scenario.getExecutionStatus().toString() + "," +
-                                    scenario.getExecutionTime() + "\n";
-                                writer.write(s);
-                            }
-                        }
-                    }
-                    writer.flush();
-                    writer.close();
-                    System.exit(0);
-                    return;
-                }
-                if(message.getMessageType() == Message.MessageType.KillProcessRequest) {
-                    if(!socket.isClosed()) {
-                        socket.close();
-                    }
-                    System.exit(0);
-                    return;
-                }
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+    public static void main(String[] args) throws Exception {
+        ServiceHandler handler = new ServiceHandler();
+        Server server = ServerBuilder.forPort(0)
+                .addService(handler)
+                .executor(Executors.newSingleThreadExecutor())
+                .build();
+        handler.addServer(server);
+        server.start();
+        int port = server.getPort();
+        System.out.println("Listening on port:" + port);
+        server.awaitTermination();
+        System.exit(0);
```

Comment:
do we need this? I think by default it would `exit(0)`. I understand that we had it earlier.

Reply:
Done